purchase_sale_stock_inter_company/tests/test_inter_company_purchase_sale_stock.py

```diff
@@ -37,32 +57,383 @@ def setUpClass(cls):
+        )
+        self.assertEqual(
+            so_lots.mapped("name"),
+            po_lots.mapped("name"),
+            msg="The lots should have the same name in both moves",
+        )
+        self.assertIn(
+            serial_3_company_a,
+            po_lots,
+            msg="Serial 333 already existed, a new one shouldn't have been created",
+        )
+    
+    def test_update_open_sale_order(self):
+        """
+        When the purchase user request extra product, the sale order gets synched if
+        it's open.
+        """
+        purchase = self._create_purchase_order(
+            self.partner_company_b, self.consumable_product
+        )
+        purchase.picking_type_id = self.warehouse_a.in_type_id
+        sale = self._approve_po(purchase)
+        sale.action_confirm()
+        # Now we add an extra product to the PO and it will show up in the SO
+        po_form = Form(purchase)
+        with po_form.order_line.new() as line:
+            line.product_id = self.consumable_product_2
+            line.product_qty = 6
+        po_form.save()
+        # It's synched and the values match
+        synched_order_line = sale.order_line.filtered(
+            lambda x: x.product_id == self.consumable_product_2
+        )
+        self.assertTrue(
+            bool(synched_order_line),
+            "The line should have been created in the sale order",
+        )
+        self.assertEqual(
+            synched_order_line.product_uom_qty,
+            6,
+            "The quantity should be equal to the one set in the purchase order",
+        )
+        # Also the moves match as well
+        so_picking_id = sale.picking_ids
+        synched_move = so_picking_id.move_lines.filtered(
+            lambda x: x.product_id == self.consumable_product_2
+        )
+        self.assertTrue(
+            bool(synched_move),
+            "The move should have been created in the delivery order",
+        )
+        self.assertEqual(
+            synched_move.product_uom_qty,
+            6,
+            "The quantity should be equal to the one set in the purchase order",
+        )
+        # The quantity is synched as well
+        purchase_line = purchase.order_line.filtered(
+            lambda x: x.product_id == self.consumable_product_2
+        ).sudo()
+        purchase_line.product_qty = 8
+        self.assertEqual(
+            synched_order_line.product_uom_qty,
+            8,
+            "The quantity should be equal to the one set in the purchase order",
+        )
+        self.assertEqual(
+            synched_move.product_uom_qty,
+            8,
+            "The quantity should synched to the one set in the purchase order",
+        )
+        # Let's decrease the quantity
+        purchase_line.product_qty = 3
+        self.assertEqual(
+            synched_order_line.product_uom_qty,
+            3,
+            "The quantity should decrease as it was in the purchase order",
+        )
+        self.assertEqual(
```

Comment:
```
2024-08-29 13:34:41,471 2528418 ERROR skata odoo.addons.purchase_sale_stock_inter_company.tests.test_inter_company_purchase_sale_stock: FAIL: TestPurchaseSaleStockInterCompany.test_update_open_sale_order
Traceback (most recent call last):
  File "/home/ntsirintanis/projects/callista/auto/addons/purchase_sale_stock_inter_company/tests/test_inter_company_purchase_sale_stock.py", line 336, in test_update_open_sale_order
    self.assertEqual(
AssertionError: 3.0 != 8 : The quantity should remain as it was as it can't be decreased from the SO

```